sdk/core/azure-core/tests/test_pipeline.py

```diff
@@ -149,6 +150,19 @@ def test_retry_types(self):
         backoff_time = retry_policy.get_backoff_time(settings)
         assert backoff_time == 4
 
+    def test_retry_after(self):
+        retry_policy = RetryPolicy()
+        request = HttpRequest("GET", "https://bing.com")
+        response = HttpResponse(request, None)
+        response.headers["retry-after-ms"] = 1000
+        pipeline_response = PipelineResponse(request, response, None)
+        retry_after = retry_policy.get_retry_after(pipeline_response)
+        assert retry_after == 1
+        response.headers.pop("retry-after-ms")
+        response.headers["Retry-After"] = 1000
+        retry_after = retry_policy.get_retry_after(pipeline_response)
+        assert retry_after == 1000
```

Comment:
Would use `pytest.mark.parametrize` to vary this over a range of values: 0, <1000, 1000, >1000

Reply:
Is it is ever possible for both headers to be supplied? If so, need a test to lock down what happens in that case.

Reply:
1000 is random number I pick. Why we want to vary the ranges?

Reply:
Now, mostly to maintain that e.g. a value of 800 ms is not rounded down to zero. But the others are just good future proofing. There is a branch in this code that does something different with a division by 1000 depending on the inputs. Tests should exercise potential corner cases around that arbitrary number to afford the greatest assurance that changes in the future don't accidentally disturb things.

Reply:
Interesting..

Our current behavior is:
if "Retry-After" == "0.8", we return 0 because we call int("Retry-After").

Reply:
Well, TBH its unclear what the actual requirements are. The code path for `"Retry-After"` clearly assumes integer (seconds). And maybe that is correct for that header.  But for any thing called "retry-after-ms" I would by default assume that I could provide 800 and that that would be interpreted as 800ms.

Reply:
Totally agree. I will ask Johan for clarification.

Reply:
Talked to Laurent. We will use float instead of int for the sleeping delay